test/helpers/foundry/foundryup.mts

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env -S node --require "./node_modules/tsx/dist/preflight.cjs" --import "./node_modules/tsx/dist/loader.mjs"
+
+import { join, relative } from 'node:path';
+import { homedir } from 'node:os';
+import { Dir, readFileSync } from 'node:fs';
+import { opendir, symlink, unlink, copyFile, rm } from 'node:fs/promises';
+import { createHash } from 'node:crypto';
+import { exit, cwd } from 'node:process';
+import { parse as parseYaml } from 'yaml';
+import {
+  BinFormat,
+  Platform,
+  extractFrom,
+  getVersion,
+  printBanner,
+  say,
+  parseArgs,
+  isCodedError,
+  noop,
+} from './helpers.mts';
+
+const parsedArgs = parseArgs();
+
```

Comment:
maybe we could split the different logic into smaller functions? From chatgpt could be something like:

- getCacheDirectory
- getBinaryArchiveUrl
- cleanCache
- checkAndDownloadBinaries
- installBinaries
- main

Reply:
Sure, most of those make sense.

I do think over-abstraction is a real concern, and can lead to reduce readability in an effort to improve it. Like `cleanCache`, for example, is already a short bit of code that is straight forward to read, moving the logic to its own file and function would increase the lines of code from 1 to about 5 (including the import and calling the function, and maybe more if adding JSDoc comments).

Feel free to add functions where you see fit, just don't blindly listen to all of ChatGPTs recommendations, I think it goes a bit too far sometimes.

Reply:
I see, your point make sense to me! I've added a proposal for split in this PR, which also add unit tests. Let me know if this makes sense, or if you prefer to have it encapsulated with less functions :pray: 

https://github.com/MetaMask/metamask-extension/pull/28833